### PR TITLE
Add support to inbox style push notifications for Android

### LINKED ISSUE #2812: Inbox style push notifications
Currently the newest push notification overrides the last push notification.

On iOS I don't believe you have much choice in the matter.  But on Android we have a couple of different types of notifications to allow stacking of notifications.
### Threading

You would increment an id every message and it would let you show multiple messages.  This would do for single threaded conversations. 

Just listing to say hey we do have options :smile:
### Inbox Style

This is the style I propose.
It groups messages by the title

Looking something like this:

We would have the contents like:
Title: server / channel
Body: Message contents
Style: 'inbox'

Because the Title would be the same for multiple messages in a thread... and we have style: inbox.  Android will auto group the messages.

Some info on sending: https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#inbox-stacking

For any Android user this is likely what they are used to.

I've opened a ticket over at raix:push to allow this: https://github.com/raix/push/issues/196

### PR DESCRIPTION
**Pull request by sampaiodiego**
@RocketChat/core 

Closes #2812

I had to update the raix:push package to `3.0.3-rc.7` because it [was not included on 3.0.2](https://github.com/RocketChat/Rocket.Chat/issues/2812#issuecomment-232382040)

I'm using the `android-chrome 192x192 (png)` asset as the server icon on the push notification. if someone has an suggest/objection on that, please adivise.
_edit: I've changed to use the 192x192 icon instead of 512x512 after a few tests._

This is how the new notifications should look like:

- Single push notification on a channel:
image

- Multiple notifications for same channel:
image

- Single direct message notification:
image

- Multiple notifications for the direct message:
image

- Mixed notifications:
image

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
Yes!! :grin:

**Comment by sampaiodiego** [comment]
jfyi the heroku instance is working and sending the new notifications: https://rocket-chat-pr-5504.herokuapp.com/

we need to update the push plugin on the cordova app, after that we can start using HTML on notifications texts for making thinks even better :D

**Comment by geekgonecrazy** [comment]
@RocketChat/android this might be of interest to you guys too.  Idk if there will be any action needed to handle.

**Comment by laggedHero** [comment]
@geekgonecrazy I'm in touch with @sampaiodiego on this one. Let's run a test and see what happens (maybe a NPE?? xD)

File: packages/rocketchat-lib/server/lib/PushNotification.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			title = `${roomName}`;
+			message = `${username}: ${message}`;
+		} else {
+			title = `${username}`;
+		}
+		let icon = RocketChat.settings.get('Assets_favicon_192').url || RocketChat.settings.get('Assets_favicon_192').defaultUrl;
+		const config = {
+			from: 'push',
+			badge: 1,
+			sound: 'chime',
```
Comment (geekgonecrazy) [inline packages/rocketchat-lib/server/lib/PushNotification.js]: is this something we had before?   If not this might actually add a sound?

Comment (sampaiodiego) [inline packages/rocketchat-lib/server/lib/PushNotification.js]: yes, look at [this](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-lib/server/lib/sendNotificationsOnMessage.js#L166)

Comment (sampaiodiego) [inline packages/rocketchat-lib/server/lib/PushNotification.js]: if we add our own sounds to the mobile apps, we can change the name of the here ;)

Comment (geekgonecrazy) [inline packages/rocketchat-lib/server/lib/PushNotification.js]: Awesome! :+1:

**Comment by engelgabriel** [comment]
@rodrigok can you get the raix:push 3.0.3 published to Atmosphere?

**Comment by rodrigok** [comment]
@engelgabriel I can't!

**Pull request merged**